### PR TITLE
fix landscape orientation for ios

### PR DESCRIPTION
**Pull request by rfrancel**
@gjn I found a way to fix the orientation of the arrow according to the orientation of the device with ios, but I couldn't find something for the other os.. Since I am stuck with this for some time, I create a PR for this and try to come back to the otehr os later.
Linked to: https://github.com/geoadmin/mf-geoadmin3/issues/3204

### CODE REVIEW

File: src/components/geolocation/GeolocationDirective.js [hunk 1]
```diff
@@ -109,8 +109,25 @@ goog.require('ga_throttle_service');
         // Get heading depending on devices
         var headingFromDevices = function() {
           var hdg = deviceOrientation.getHeading();
+          var orientation = window.orientation;
```
Comment (gjn) [inline src/components/geolocation/GeolocationDirective.js]: Please you the $window wrapper. (It should already be there)

**Comment by gjn** [comment]
Please provide a testlink. @mariavkourouni Could you test it?

**Comment by rfrancel** [comment]
test link: https://mf-geoadmin3.dev.bgdi.ch/frr_geolocation_test/
or:
image

File: src/components/geolocation/GeolocationDirective.js [hunk 2]
```diff
@@ -109,8 +109,25 @@ goog.require('ga_throttle_service');
...
           if (!gaBrowserSniffer.ios) {
-            hdg = -hdg;
+            hdg = (2 * Math.PI) - hdg;
+            }
```
Comment (oterral) [inline src/components/geolocation/GeolocationDirective.js]: indent

File: src/components/geolocation/GeolocationDirective.js [hunk 3]
```diff
@@ -109,8 +109,25 @@ goog.require('ga_throttle_service');
...
+          switch (orientation) {
+            case -90:
+              hdg = hdg - (Math.PI / 2);
+            break;
```
Comment (oterral) [inline src/components/geolocation/GeolocationDirective.js]: indent

File: src/components/geolocation/GeolocationDirective.js [hunk 4]
```diff
@@ -109,8 +109,25 @@ goog.require('ga_throttle_service');
...
+
+            case 180:
+              hdg = hdg + Math.PI;
+            break;
```
Comment (oterral) [inline src/components/geolocation/GeolocationDirective.js]: indent

File: src/components/geolocation/GeolocationDirective.js [hunk 5]
```diff
@@ -109,8 +109,25 @@ goog.require('ga_throttle_service');
...
+
+            case 90:
+              hdg = hdg + (Math.PI / 2);
+            break;
```
Comment (oterral) [inline src/components/geolocation/GeolocationDirective.js]: indent

**Comment by oterral** [comment]
@rfrancel did you try the property window.screen.orientation for firefox and chrome 
see http://caniuse.com/#feat=screen-orientation

File: src/components/geolocation/GeolocationDirective.js [hunk 6]
```diff
@@ -109,9 +109,11 @@ goog.require('ga_throttle_service');
         // Get heading depending on devices
         var headingFromDevices = function() {
           var hdg = deviceOrientation.getHeading();
-          var orientation = window.orientation;
+          var orientation = $window.orientation ||
+              $window.screen.orientation.angle;
```
Comment (oterral) [inline src/components/geolocation/GeolocationDirective.js]: This will trigger an error on IE. You ned to test if `$window.screen.orientation` exists first

**Comment by oterral** [comment]
For IE mobile you should be able to use [`screen.msOrientation` ](https://msdn.microsoft.com/en-us/library/dn433241(v=vs.85).aspx)

File: src/components/geolocation/GeolocationDirective.js [hunk 7]
```diff
@@ -109,8 +109,27 @@ goog.require('ga_throttle_service');
         // Get heading depending on devices
         var headingFromDevices = function() {
           var hdg = deviceOrientation.getHeading();
+          var orientation = $window.orientation ||
+              $window.screen.orientation.angle;
           if (!gaBrowserSniffer.ios) {
-            hdg = -hdg;
+            hdg = (2 * Math.PI) - hdg;
+            //orientation = screen.orientation.angle;
+            }
```
Comment (oterral) [inline src/components/geolocation/GeolocationDirective.js]: indent

**Comment by rfrancel** [comment]
@oterral  This fixes orientation heading for ios browsers and chrome on android. On chrome windows (surface at least), geolocation's arrow does not move at all, stays pointing to the north..
msOrientation does only give an indicaton about lanscape and portrait, so it is device specific. It does not give the angle relative to natural device orientation..
I think I have corrected the indents..

**Comment by gjn** [comment]
Please rebase.

**Pull request merged**